src/main/java/net/sf/jabref/gui/ImportInspectionDialog.java

```diff
@@ -537,7 +538,7 @@ private void generateKeys() {
         for (BibEntry entry : entries) {
             database.removeEntry(entry);
         }
-        entries.getReadWriteLock().writeLock().lock();
+        entries.getReadWriteLock().writeLock().unlock();
```

Comment:
Can you create a try-finally with unlocking the lock in the finally part? This should be done in the other part of the code as well.